### PR TITLE
Bump pandas and sqlalchemy

### PR DESCRIPTION
**Pull request by surister**
## Summary of the changes / Why this is an improvement

Update pandas and sqlalchemy to latest (or fairly modern versions)

## Checklist

 - [x] Link to issue this PR refers to (if applicable): https://github.com/crate/cratedb-examples/issues/387
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by surister** [comment]
Found some issues with Pandas while running all the notebooks, will try to fix later

**Review by amotl — COMMENTED** [review]
Thanks a stack for working on this. There is an error on CI wrt. to PyCaret. I hope it will turn out well. Maybe PyCaret exclusively needs pandas<2 or SQLAlchemy<2?

```python
File /opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/pycaret/internal/memory.py:25
ImportError: cannot import name '_format_load_msg' from 'joblib.memory' (/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/joblib/memory.py)
```

-- https://github.com/crate/cratedb-examples/actions/runs/8604266577/job/23577980009?pr=400#step:6:859

If this is the case, we may need to split up the ingredients to be separated by pandas/SQLAlchemy major versions in one way or another, either by providing dedicated `requirements.txt` files, or by separating notebooks into different folders.

File: topic/timeseries/dask-weather-data-import.ipynb [hunk 1]
```diff
@@ -155,163 +175,16 @@
...
...>\n      <td>0.0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>41515</td>\n      <td>Asadabad</td>\n      <td>1957-07-03</td>\n      <td>Summer</td>\n      <td>24.3</td>\n      <td>16.7</td>\n      <td>35.6</td>\n      <td>1.0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>41515</td>\n      <td>Asadabad</td>\n      <td>1957-07-04</td>\n      <td>Summer</td>\n      <td>26.6</td>\n      <td>16.1</td>\n      <td>37.8</td>\n      <td>4.1</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>41515</td>\n      <td>Asadabad</td>\n      <td>1957-07-05</td>\n      <td>Summer</td>\n      <td>30.8</td>\n      <td>20.0</td>\n      <td>41.7</td>\n      <td>0.0</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
```
[the remark this answers is not in the document]
Comment (surister) [reply topic/timeseries/dask-weather-data-import.ipynb:185]: I might have cleaned up some cell's output, will fix later

**Comment by amotl** [comment]
There is an error on CI wrt. to PyCaret: `ImportError: cannot import name '_format_load_msg' from 'joblib.memory'`.

I discovered the same error here:
- https://github.com/crate-workbench/mlflow-cratedb/pull/124

This adjustment aims to mitigate it:
https://github.com/crate-workbench/mlflow-cratedb/pull/124/commits/49cb2b61132b19be2c12d425d9f83e6f5729c6ea

**Comment by amotl** [comment]
> There is an error with PyCaret.

There is also a flaw with Dask and most recent Python 3.11.9. 🐛 

https://github.com/pycaret/pycaret/issues/3960#issuecomment-2045881422

**Comment by amotl** [comment]
The TSML stack modulo LLM/LangChain needed a fix. After rebasing, relevant obstacles should be remedied for you.

- GH-402

**Comment by amotl** [comment]
More havocs caused by recent updates had to be remedied, to make relevant artefacts work on Google Colab again. Thanks for your support, @marijaselakovic.

- GH-403
- GH-404

@surister: You may now want to rebase again, to be in a better position than before.

**Comment by amotl** [comment]
This patch is currently blocked on errors like this, on behalf of GH-430:
```
TypeError: Could not convert string 'BerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlinBerlin' to numeric
```

-- https://github.com/crate/cratedb-examples/actions/runs/8975962618/job/24651748395?pr=430#step:6:1839

**Pull request closed without merging**